### PR TITLE
improvement(Confirmation): Confirmation is not a strategy.

### PR DESCRIPTION
**Pull request by jimsynz**
Well, actually it is - but conceptually it's not.

### CODE REVIEW

File: lib/ash_authentication/add_ons/confirmation.ex [hunk 1]
```diff
@@ -0,0 +1,140 @@
+defmodule AshAuthentication.AddOn.Confirmation do
+  import AshAuthentication.Dsl
+
+  @moduledoc """
+  Confirmation support.
+
+  Sometimes when creating a new user, or changing a sensitive attribute (such as
+  their email address) you may want to for the user to confirm by way of sending
```
Comment (rellen) [inline lib/ash_authentication/add_ons/confirmation.ex:8]: I think there's a typo here???
```suggestion
  their email address) you may want wait to for the user to confirm by way of sending
```

File: lib/ash_authentication/add_ons/confirmation/transformer.ex [hunk 2]
```diff
@@ -49,8 +49,8 @@ defmodule AshAuthentication.Strategy.Confirmation.Transformer do
       [_ | _] ->
         {:error,
          DslError.exception(
-           path: [:authentication, :strategies, :confirmation],
-           message: "Multiple confirmation strategies are not supported"
+           path: [:authentication, :add_ons, :confirmation],
+           message: "Multiple confirmation add ons are not supported"
```
Comment (zachdaniel) [inline lib/ash_authentication/add_ons/confirmation/transformer.ex:53]: unrelated to this PR, but why not? Is it routing? If we put the add_on name in the route, we could support any number of confirmations, right?

Comment (jimsynz) [reply lib/ash_authentication/add_ons/confirmation/transformer.ex:53]: yeah, that occurred to me too. seems like an arbitrary decision I made at the start

File: lib/ash_authentication/add_ons/confirmation.ex (hunk 1 above)
Comment (jimsynz) [reply lib/ash_authentication/add_ons/confirmation.ex:8]: Nice catch!

File: lib/ash_authentication/add_ons/confirmation.ex [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+          name: :confirm
+        }
+
+  @doc """
+  Generate a confirmation token for a changeset.
+
+  This will generate a token with the `"act"` claim set to the confirmation
+  action for the strategy, and the `"chg"` claim will contain any changes.
+
+  FIXME: The "chg" claim should encrypt the contents of the changes so as to not
```
Comment (zachdaniel) [inline lib/ash_authentication/add_ons/confirmation.ex:121]: we should make a GH issue for this, so this isn't lost.

**Review by zachdaniel — APPROVED** [review]

**Review by rellen — APPROVED** [review]
just one doco typo found.  I am not across the particulars to review the code changes

File: lib/ash_authentication/add_ons/confirmation.ex (hunk 3 above)
Comment (jimsynz) [reply lib/ash_authentication/add_ons/confirmation.ex]: :+1:

**Pull request merged**